From 30b45b5d61f0e4f13c84da5aba1a0d0f4b2e65a7 Mon Sep 17 00:00:00 2001
From: Dinesh Sudham <dineshx.sudham@intel.com>
Date: Wed, 10 Jun 2020 18:49:01 +0800
Subject: [PATCH] DRVLIB_SW-3656:GSWAPI: memory leak fixes in switch_api

Temorary workaround for fix memory leaks warings from klockwork
first member not intailized and maybe lost we return
from function, keep intialized member as first structure to clear
warings

Signed-off-by: Dinesh Sudham <dineshx.sudham@intel.com>
---
 drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.h    | 4 ++--
 drivers/net/ethernet/lantiq/switch-api/mac/gswss_api.h    | 2 +-
 drivers/net/ethernet/lantiq/switch-api/mac/xgmac_common.h | 3 +--
 3 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.h b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.h
index 801bc717549a..ddb435bb9feb 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.h
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_flow_core.h
@@ -923,6 +923,8 @@ typedef struct {
 } GSW_PCE_RuleManage_t;
 
 typedef struct {
+	/**Switch Opertations**/
+	struct core_ops ops;
 	gsw_devtype_t	sdev;
 	port_config_t pconfig[MAX_PORT_NUMBER];
 	avlan_tbl_t avtable[VLAN_ACTIVE_TABLE_SIZE];
@@ -1036,8 +1038,6 @@ typedef struct {
 	/*IRQ*/
 	struct pce_irq_linklist *PceIrqList;
 	u32  irq_num;
-	/**Switch Opertations**/
-	struct core_ops ops;
 } ethsw_api_dev_t;
 
 
diff --git a/drivers/net/ethernet/lantiq/switch-api/mac/gswss_api.h b/drivers/net/ethernet/lantiq/switch-api/mac/gswss_api.h
index 451d90d898ec..3077095b96d1 100644
--- a/drivers/net/ethernet/lantiq/switch-api/mac/gswss_api.h
+++ b/drivers/net/ethernet/lantiq/switch-api/mac/gswss_api.h
@@ -184,6 +184,7 @@ enum {
 struct adap_prv_data;
 
 struct adap_prv_data {
+	struct adap_ops ops;
 	u32 flags;
 #ifdef __KERNEL__
 	/* Adaption layer private data */
@@ -194,7 +195,6 @@ struct adap_prv_data {
 	u32 ss_addr_base;
 #endif
 	u32 core_en_cnt;
-	struct adap_ops ops;
 	/* Mac Cli */
 	GSW_MAC_Cli_t *mac_cli;
 };
diff --git a/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_common.h b/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_common.h
index bf3887595f9c..e06f082b3708 100644
--- a/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_common.h
+++ b/drivers/net/ethernet/lantiq/switch-api/mac/xgmac_common.h
@@ -345,6 +345,7 @@ struct mac_irq_hdl {
 struct mac_prv_data;
 
 struct mac_prv_data {
+	struct mac_ops ops;
 
 	/* XGMAC registers for indirect accessing */
 	u32 xgmac_ctrl_reg;
@@ -554,8 +555,6 @@ struct mac_prv_data {
 
 	/* MAC IRQ handler */
 	struct mac_irq_hdl *irq_hdl;
-
-	struct mac_ops ops;
 };
 
 extern struct mac_prv_data prv_data[10];
